### PR TITLE
update(IText): Add method _enterEditing

### PR DESCRIPTION
**Pull request by zhe-he**
Add method _enterEditing to prevent triggering text:editing:entered.

https://github.com/fabricjs/fabric.js/issues/10187

### CODE REVIEW

File: src/shapes/IText/ITextBehavior.ts [hunk 1]
```diff
@@ -387,6 +387,22 @@ export abstract class ITextBehavior<
...
+        e,
+      });
+      this.canvas.requestRenderAll();
+    }
+  }
+
+  /**
+   * runs the actual logic that enter from editing state, see {@link enterEditing}
+   */
+  protected _enterEditing() {
```
Comment (asturur) [inline src/shapes/IText/ITextBehavior.ts:405]: if is protected you can't really call it by yourself. This should stay public and have a different name.
Maybe enterEditingImpl ( as for implementation ) is better than the underscore prefix?

**Comment by asturur** [comment]
Also please divide eventual tests for enterEditing for the 2 respective methods, under JEST.
If you don't know how to do that i can provide guidance.

File: src/shapes/IText/ITextBehavior.ts (hunk 1 above)
Comment (zhe-he) [reply src/shapes/IText/ITextBehavior.ts]: I found that other internal methods that do not trigger the fire event are all prefixed with an underscore, such as 
 _discardActiveObject/discardActiveObject and _exitEditing/exitEditing, which is why I named it this way.
I should also change the _exitEditing method to public.

Comment (zhe-he) [reply src/shapes/IText/ITextBehavior.ts]: To unify, I changed the method name _exitEditing to exitEditingImpl.

**Comment by zhe-he** [comment]
Test cases have been added, but I cannot test them locally. The reasons are as follows:
When I executed `./node_modules/.bin/jest ./src/shapes/IText/ITextBehavior.test.ts` before adding the test cases, the previous test cases reported errors. This issue may need your attention.

**Comment by asturur** [comment]
Hopefully i didn't break anything.
I left _exitEditing to avoid breaking changes and introduced anyway exitEditingImpl

**Comment by asturur** [comment]
> Test cases have been added, but I cannot test them locally. The reasons are as follows: When I executed `./node_modules/.bin/jest ./src/shapes/IText/ITextBehavior.test.ts` before adding the test cases, the previous test cases reported errors. This issue may need your attention.

There was an unrestored mock in the previous tests

**Comment by zhe-he** [comment]
I looked at the v5 code and found that the _exitEditing method was not present in the previous version; it was added in v6. Therefore, removing it will not affect the old version. Additionally, in v6, it is marked as a private method, and if used externally, TypeScript will issue a warning unless used as shown below. This is why I removed it.

image
image

**Comment by asturur** [comment]
Yes but you never know what people do.
I can't say 'typescript is warning you so is not a breaking change' if someone override that method in a subclass could have troubles.

**Comment by asturur** [comment]
Easier to keep and remove later.

**Pull request merged**